### PR TITLE
Wire up caas operator logging to controller

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

As well as logging to standard k8s logger, we also send logs to the controller. This conforms with how other agents are expected to operate.

## QA steps

deploy k8s charm
run juju debug-log on the k8s model

### CODE REVIEW

File: worker/caasoperator/caasoperator.go [hunk 1]
```diff
@@ -429,7 +429,7 @@ func charmModified(local LocalState, remote remotestate.Snapshot) bool {
 
 func (op *caasOperator) ensureCharm() (*charm.URL, int, error) {
 	charmDir := op.paths.GetCharmDir()
-	if _, err := os.Stat(charmDir); !os.IsNotExist(err) {
+	if _, err := os.Stat(charmDir); err != nil && !os.IsNotExist(err) {
```
Comment (babbageclunk) [inline worker/caasoperator/caasoperator.go]: Do you really need the `err != nil` here - doesn't `os.IsNotExist` return false for nil? I tried it in the playground and it worked.

Comment (wallyworld) [reply worker/caasoperator/caasoperator.go]: !os.IsNotExist(err) returns true for nil err and thus returns without doing the download. and worse, the returned error is nil and so the caller thinks there's nothing wrong but actually nothing was downloaded

Comment (babbageclunk) [reply worker/caasoperator/caasoperator.go]: Ok, I get this now you explained it in IRC.

Comment (wallyworld) [reply worker/caasoperator/caasoperator.go]: Actually,the original code was correct after all. The operator was too simplistic in how it determined if a charm already existed. I needed to adopt the code already used in the IAAS uniter.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
[fix operator charm download - use code from uniter, add operator state](https://trello.com/c/GffhSfle/448-fix-operator-charm-download-use-code-from-uniter-add-operator-state)

**Comment by wallyworld** [comment]
[wire up k8s operator logging to log to controller](https://trello.com/c/Y8TQ9YRZ/447-wire-up-k8s-operator-logging-to-log-to-controller)

**Pull request merged**